crates/telio-model/src/api_config.rs

```diff
@@ -77,65 +78,127 @@ impl FeatureWireguard {
     /// The unique identifier of the device, used for meshnet ID
     pub fingerprint: String,
     /// Heartbeat interval in seconds. Default value is 3600.
+    #[serde_as(as = "DurationSeconds<u64>")]
     #[serde(default = "FeatureNurse::get_default_heartbeat_interval")]
-    pub heartbeat_interval: Option<u32>,
+    pub heartbeat_interval: Duration,
     /// Initial heartbeat interval in seconds. Default value is 300.
+    #[serde_as(as = "DurationSeconds<u64>")]
     #[serde(default = "FeatureNurse::get_default_initial_heartbeat_interval")]
-    pub initial_heartbeat_interval: Option<u32>,
-    /// QoS configuration for Nurse
+    pub initial_heartbeat_interval: Duration,
+    /// QoS configuration for Nurse. Enabled by default.
+    #[serde(default = "FeatureNurse::get_default_qos")]
     pub qos: Option<FeatureQoS>,
-    /// Enable/disable collecting nat type
-    pub enable_nat_type_collection: Option<bool>,
-    /// Enable/disable Relay connection data
+    /// Enable/disable collecting nat type. Disabled by default.
+    #[serde(default = "FeatureNurse::get_default_enable_nat_type_collection")]
+    pub enable_nat_type_collection: bool,
+    /// Enable/disable Relay connection data. Enabled by default.
     #[serde(default = "FeatureNurse::get_default_enable_relay_conn_data")]
-    pub enable_relay_conn_data: Option<bool>,
-    /// Enable/disable NAT-traversal connections data
+    pub enable_relay_conn_data: bool,
+    /// Enable/disable NAT-traversal connections data. Enabled by default.
     #[serde(default = "FeatureNurse::get_default_enable_nat_traversal_conn_data")]
-    pub enable_nat_traversal_conn_data: Option<bool>,
+    pub enable_nat_traversal_conn_data: bool,
 }
 
 impl FeatureNurse {
-    /// One hour
-    pub const DEFAULT_HEARTBEAT_INTERVAL: Duration = Duration::from_secs(3600);
-    /// 5 minutes
-    pub const DEFAULT_INITIAL_HEARTBEAT_INTERVAL: Duration = Duration::from_secs(300);
+    fn get_default_heartbeat_interval() -> Duration {
+        Duration::from_secs(60 * 60)
+    }
 
-    fn get_default_enable_relay_conn_data() -> Option<bool> {
-        Some(true)
+    fn get_default_initial_heartbeat_interval() -> Duration {
+        Duration::from_secs(5 * 60)
     }
 
-    fn get_default_enable_nat_traversal_conn_data() -> Option<bool> {
-        Some(true)
+    fn get_default_qos() -> Option<FeatureQoS> {
+        Some(Default::default())
     }
 
-    fn get_default_heartbeat_interval() -> Option<u32> {
-        // 3600 seconds
-        Some(Self::DEFAULT_HEARTBEAT_INTERVAL.as_secs() as u32)
+    fn get_default_enable_nat_type_collection() -> bool {
+        false
     }
 
-    fn get_default_initial_heartbeat_interval() -> Option<u32> {
-        // 300 seconds
-        Some(Self::DEFAULT_INITIAL_HEARTBEAT_INTERVAL.as_secs() as u32)
+    fn get_default_enable_relay_conn_data() -> bool {
+        true
+    }
+
+    fn get_default_enable_nat_traversal_conn_data() -> bool {
+        true
+    }
```

Comment:
Nitpick: I'm voting for stripping these `get_` prefixes

Reply:
+1

Reply:
👍